File: jsfs-ser/cs/src-ser/com/wilutions/jsfs/DispatcherService.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using byps;

namespace com.wilutions.jsfs
{
	
	/// <summary>
	/// This class defines the interface of the JSFS Dispatcher.
	/// </summary>
	/// <remarks>
	/// The functions in this interface have to be called with a token obtained from an
	/// authentication process with the token service of "Your Web Application".
	/// </remarks>
	// checkpoint byps.gen.cs.GenRemoteClass:93
	public interface DispatcherService : BRemote {
	
		/// <summary>
		/// Register a file system service.
		/// </summary>
		/// <remarks>
		/// This function is called from the JSFS Agent when it starts.
		/// The JSFS Dispatcher maps the given token on the given service. The browser
		/// requests this service by a call to {@link #getService(String, boolean)}.
		/// </remarks>
		void RegisterService(String token, FileSystemService service);
		void RegisterService(String token, FileSystemService service, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Register a file system service.
		/// </summary>
		/// <remarks>
		/// This function is called from the JSFS Agent when it starts.
		/// The JSFS Dispatcher maps the given token on the given service. The browser
		/// requests this service by a call to {@link #getService(String, boolean)}.
		/// </remarks>
		Task RegisterServiceAsync(String token, FileSystemService service);
		
		/// <summary>
		/// Unregister a file system service.
		/// </summary>
		/// <remarks>
		/// This function is called from the JSFS Agent if it terminates.
		/// </remarks>
		void UnregisterService(String token);
		void UnregisterService(String token, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Unregister a file system service.
		/// </summary>
		/// <remarks>
		/// This function is called from the JSFS Agent if it terminates.
		/// </remarks>
		Task UnregisterServiceAsync(String token);
		
		/// <summary>
		/// Get the file system service for the given token.
		/// </summary>
		FileSystemService GetService(String token, bool onlyHere);
		void GetService(String token, bool onlyHere, BAsyncResult<FileSystemService> asyncResult) ;
		/// <summary>
		/// Get the file system service for the given token.
		/// </summary>
		Task<FileSystemService> GetServiceAsync(String token, bool onlyHere);
		
		/// <summary>
		/// Register a notify service.
		/// </summary>
		/// <remarks>
		/// The notify service is implemented by JavaScript inside the web page. The JSFS Agent requests
		/// the notify service in order to send messages to the browser.
		/// </remarks>
		void RegisterNotifyService(String token, FileSystemNotify service);
		void RegisterNotifyService(String token, FileSystemNotify service, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Register a notify service.
		/// </summary>
		/// <remarks>
		/// The notify service is implemented by JavaScript inside the web page. The JSFS Agent requests
		/// the notify service in order to send messages to the browser.
		/// </remarks>
		Task RegisterNotifyServiceAsync(String token, FileSystemNotify service);
		
		/// <summary>
		/// Unregister a notify service.
		/// </summary>
		/// <remarks>
		/// This function should be called by the browser when the web page is unloaded.
		/// </remarks>
		void UnregisterNotifyService(String token);
		void UnregisterNotifyService(String token, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Unregister a notify service.
		/// </summary>
		/// <remarks>
		/// This function should be called by the browser when the web page is unloaded.
		/// </remarks>
		Task UnregisterNotifyServiceAsync(String token);
		
		/// <summary>
		/// Get the notify service interface for a given token.
		/// </summary>
		FileSystemNotify GetNotifyService(String token, bool onlyHere);
		void GetNotifyService(String token, bool onlyHere, BAsyncResult<FileSystemNotify> asyncResult) ;
		/// <summary>
		/// Get the notify service interface for a given token.
		/// </summary>
		Task<FileSystemNotify> GetNotifyServiceAsync(String token, bool onlyHere);
		
		/// <summary>
		/// Keep the token-to-services-associations alive.
		/// </summary>
		/// <remarks>
		/// Keeps the application server session alive. If the session has already been invalidated,
		/// the BYPS framework triggers a re-login. This will cause an authentication with "Your Web Application"
		/// and the services will be registered again.
		/// </remarks>
		void KeepAlive(String token);
		void KeepAlive(String token, BAsyncResult<Object> asyncResult) ;
		/// <summary>
		/// Keep the token-to-services-associations alive.
		/// </summary>
		/// <remarks>
		/// Keeps the application server session alive. If the session has already been invalidated,
		/// the BYPS framework triggers a re-login. This will cause an authentication with "Your Web Application"
		/// and the services will be registered again.
		/// </remarks>
		Task KeepAliveAsync(String token);
		
		
	}
}  // end namespace

File: jsfs-api/src/com/wilutions/jsfs/DispatcherService.java
/* USE THIS FILE ACCORDING TO THE COPYRIGHT RULES IN LICENSE.TXT WHICH IS PART OF THE SOURCE CODE PACKAGE */ 
package com.wilutions.jsfs;

import byps.BRemote;
import byps.RemoteException;


/**
 * This class defines the interface of the JSFS Dispatcher.
 * The functions in this interface have to be called with a token obtained from an 
 * authentication process with the token service of "Your Web Application".
 */
public interface DispatcherService extends BRemote {
  
  /**
   * Register a file system service.
   * This function is called from the JSFS Agent when it starts. 
   * The JSFS Dispatcher maps the given token on the given service. The browser
   * requests this service by a call to {@link #getService(String, boolean)}.
   * @param token Token obtained from authentication.
   * @param service Service object.
   * @throws RemoteException
   */
  public void registerService(String token, FileSystemService service) throws RemoteException;
  
  /**
   * Unregister a file system service.
   * This function is called from the JSFS Agent if it terminates.
   * @param token Token obtained from authentication.
   * @throws RemoteException
   */
  public void unregisterService(String token) throws RemoteException;
  
  /**
   * Get the file system service for the given token.
   * @param token Token obtained from authentication.
   * @param onlyHere false, if other JSFS Dispatcher services should be asked for the file system service too.
   * @return File system service.
   * @throws RemoteException
   */
  public FileSystemService getService(String token, boolean onlyHere) throws RemoteException;
  
  /**
   * Register a notify service.
   * The notify service is implemented by JavaScript inside the web page. The JSFS Agent requests
   * the notify service in order to send messages to the browser.
   * @param token Token obtained from authentication. 
   * @param service Notify service.
   * @throws RemoteException
   */
  public void registerNotifyService(String token, FileSystemNotify service) throws RemoteException;
  
  /**
   * Unregister a notify service.
   * This function should be called by the browser when the web page is unloaded.
   * @param token Token obtained from authentication.
   * @throws RemoteException
   */
  public void unregisterNotifyService(String token) throws RemoteException;
  
  /**
   * Get the notify service interface for a given token.
   * @param token Token obtained from authentication.
   * @param onlyHere false, if other JSFS Dispatcher services should be asked for the notify service too.
   * @return Notify service.
   * @throws RemoteException
   */
  public FileSystemNotify getNotifyService(String token, boolean onlyHere) throws RemoteException;
 
  /**
   * Keep the token-to-services-associations alive.
   * Keeps the application server session alive. If the session has already been invalidated,
   * the BYPS framework triggers a re-login. This will cause an authentication with "Your Web Application"
   * and the services will be registered again.
   * @param token Token obtained from authentication.
   * @throws RemoteException
   */
  public void keepAlive(String token) throws RemoteException;
  
}
